### PR TITLE
test: pretty printing IPs in test

### PR DESCRIPTION
**Pull request by alyssawilk**
Changing our tests from being of the form
 IpVersions/SslSocketTest.HalfClose/1
to
 IpVersions/SslSocketTest.HalfClose/IPv4

*Risk Level*: Low
*Testing*: n/a (manually looked at test output)
*Docs Changes*: n/a
*Release Notes*: n/a

### CODE REVIEW

**Comment by alyssawilk** [comment]
Daniel can verify correctness but @mattklein123 can you just let me know if this will mess with any code reorgs you have in flight? I can drop any arbitrary set of files from this PR, I just don't want to make your slog any sloggier :-)

**Comment by mattklein123** [comment]
I don't have any inflight right now. Let's just get this merged.

File: test/test_common/utility.h [hunk 1]
```diff
@@ -230,6 +230,15 @@ class TestUtility {
     MessageUtil::loadFromYaml(yaml, message);
     return message;
   }
+
+  // Allows pretty printed test names for TEST_P using TestEnvironment::getIpVersionsForTest()
```
Comment (danielhochman) [inline test/test_common/utility.h]: super nit: periods at end of comments

Comment (danielhochman) [reply test/test_common/utility.h]: can reapprove with change or merge, either is fine with me.

**Pull request merged**